SEP-0013.md

```diff
@@ -0,0 +1,198 @@
+The benefit of this is that the same type of metadata can be accessed in the same way irrespective
+of what data object is being used and what instrument the metadata has come from. To facilitate
+this, a set of standardized attribute names must be established for storing and accessing commonly
+used metadata in solar physics.  Then a metadata object can implement a translation from the
+original file/object containing the metadata file, e.g. FITS header, and the attributes.
+
+The purpose of this SEP is to agree the names of the different metadata attributes. To start with
+the list does not have to be exhaustive. We envision that it can be appended to over time. But
+once an attribute name is addedit should preferably not be changed so that users can rely on a
+stable, backwards compatible API.
+
+For now, this API does not include WCS metadata and should not be linked to a specific file format,
+e.g. FITS.
+
+
+# SunPy Metadata API
+Below is a class definition of an Metadata Abstract Base Class (ABC) for solar physics.
+
+```python
+
+class MetaABC():
+    ##### Instrument info #####
+
+    @abc.abstractproperty
+    def observatory(self):
+        """
+        Name of the observatory
+        """
+
+    @abc.abstractproperty
+    def instrument(self):
+        """
+        Name of the instrument
+        """
+
+    @abc.abstractproperty
+    def detector(self):
+        """
+        Name of the detector
+        """
+
+    ##### Data Info ####
+
+    @abc.abstractproperty
+    def data_processing_level(self):
+        """
+        The level to which the data has been processed.
+        """
+
+    @abc.abstractproperty
+    def data_version(self):
+        """
+        The data version.
+        """
+
+    ##### Observer Coordinate Info #####
+
+    @abc.abstractproperty
+    def observer_location(self):
+        """
+        Coordinate of observatory location.
+        """
+
+    def observer_radial_velocity(self):
+        """
+        Velocity of observer in direction of source.
+        """
+
+    @abc.abstractproperty
+    def celestial_frame(self):
+        """
+        The celestial coordinate frame of the image axes.
+        """
+
+    @abc.abstractproperty
+    def distance_to_sun(self):
+        """
+        Distance to Sun center from observatory.
+        """
+
+    @abc.abstractproperty
+    def light_travel_time(self):
+        """
+        Light travel time from emission source to observer.
+        """
+
+    @abc.abstractproperty
+    def light_travel_time_to_earth(self):
+        """
+        Light travel time from observer to Earth.
+        """
+
+    ##### Time Info #####
+
+    @abc.abstractproperty
+    def date_reference(self):
+        """
+        The base time from which time axis values are measured.
+        """
+
+    @abc.abstractproperty
+    def date_start(self):
+        """
+        Time at which the observation(s) were started.
+        """
+
+    @abc.abstractproperty
+    def date_average(self):
```

Comment:
Isn't this redundant if you have the start/end time? If not, how exactly is mean/representative defined?

Reply:
For this, think DATE_OBS in a FITS header.  It's "the" time of the observation, however that's defined by the instrument team.

However, if the data is in an `NDCube` I would argue this information should be stored in `.global_coords`.  So if we link this metadata standard with `NDCube` I think this property should be removed.  But here I haven't required that this metadata class be used with an `NDCube` instance.  Perhaps we should?